packages/astro/src/core/errors/dev/utils.ts

```diff
@@ -32,21 +40,33 @@ export function collectErrorMetadata(e: any, filePath?: URL): ErrorWithMetadata
 		}
 
 		// Generic error (probably from Vite, and already formatted)
-		if (!error.hint) {
-			error.hint = generateHint(e, filePath);
-		}
+		error.hint = generateHint(e);
 	});
 
 	// If we received an array of errors and it's not from us, it should be from ESBuild, try to extract info for Vite to display
 	if (!AggregateError.is(e) && Array.isArray((e as any).errors)) {
 		(e as BuildResult).errors.forEach((buildError, i) => {
-			const { location, pluginName } = buildError;
+			const { location, pluginName, text } = buildError;
+
+			// ESBuild can give us a slightly better error message than the one in the error, so let's use it
```

Comment:
ESBuild's errors are actually pretty amazing in general! Vite does some enhancement of them, but even by themselves they're really informative